nixos/modules/programs/shadow.nix

```
@@ -138,7 +139,7 @@ in
 
   ###### implementation
 
-  config = {
+  config = lib.mkIf config.security.shadow.enable {
```

```suggestion
  config = 
   let myAssert = a: if config.security.shadow.enable == false -> config.services.greetd.enable then a else throw "One of config.security.shadow or config.services.greetd must be set.";
   in
   myAssert lib.mkIf config.security.shadow.enable {
```


This is how I'd do it.

---

We don't want to use throw in the module system. This should be done in assertions.